enterprise/internal/campaigns/webhooks.go

```diff
@@ -764,37 +764,48 @@ func (h *BitbucketServerWebhook) Upsert(every time.Duration) {
 		}
 
 		for _, e := range es {
-			c, _ := e.Configuration()
+			err = func() error {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+
+				c, _ := e.Configuration()
+
+				con, ok := c.(*schema.BitbucketServerConnection)
+				if !ok {
+					return nil
+				}
+
+				client, err := bbs.NewClient(con, nil)
+				if err != nil {
+					return errors.Wrap(err, "creating client")
+				}
+
+				secret := con.WebhookSecret()
+				if secret == "" {
+					// Delete it in case it was created in the past
+					err = client.DeleteWebhook(ctx, h.Name)
+					if err != nil {
+						return errors.Wrap(err, "deleting webhook")
+					}
+					return nil
+				}
+
+				endpoint := externalURL() + "/.api/bitbucket-server-webhooks"
+
+				wh := bbs.Webhook{
+					Name:     h.Name,
+					Scope:    "global",
+					Events:   []string{"pr", "repo"},
+					Endpoint: endpoint,
+					Secret:   secret,
+				}
+
+				err = client.UpsertWebhook(context.Background(), wh)
```

Comment:
Why not use the `ctx` from above?

```suggestion
				err = client.UpsertWebhook(ctx, wh)
```